DisjointSet.java

```diff
@@ -0,0 +1,47 @@
+
+public class DisjointSet {
+	int[] parent;
+	int[] rank;
+
+	public DisjointSet(int numElements) {
+		parent = new int[numElements];
+		for (int i = 0; i < numElements; i++) {
+			parent[i] = i + 1;
```

Comment:
@Thomas-de-Rivaz I use i+1 here since in the array with Islands 0 is used to denote water. I will think how to improve this in order to avoid the impression that the code is copied. Thank you for pointing this out.